src/modules/jcms_rest/src/Plugin/rest/resource/JobAdvertListRestResource.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Drupal\jcms_rest\Plugin\rest\resource;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\jcms_rest\Response\JCMSRestResponse;
+use Drupal\node\Entity\Node;
+use Drupal\node\NodeInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Provides a resource to get view modes by entity and bundle.
+ *
+ * @RestResource(
+ *   id = "job_advert_list_rest_resource",
+ *   label = @Translation("Job advert list rest resource"),
+ *   uri_paths = {
+ *     "canonical" = "/job-adverts"
+ *   }
+ * )
+ */
+class JobAdvertListRestResource extends AbstractRestResourceBase {
+
+  /**
+   * Responds to GET requests.
+   *
+   * Returns a list of bundles for specified entity.
+   *
+   * @throws \Symfony\Component\HttpKernel\Exception\HttpException
+   *   Throws exception expected.
+   *
+   * @todo - elife - nlisgo - Handle version specific requests
+   */
+  public function get() {
+    $base_query = \Drupal::entityQuery('node')
+      ->condition('status', NodeInterface::PUBLISHED)
+      ->condition('changed', \Drupal::time()->getRequestTime(), '<')
+      ->condition('type', 'job_advert');
+
+    $this->filterShow($base_query);
+    $count_query = clone $base_query;
+    $items_query = clone $base_query;
+    $response_data = [
+      'total' => 0,
+      'items' => [],
+    ];
+    $nodes = [];
+    if ($total = $count_query->count()->execute()) {
+      $response_data['total'] = (int) $total;
+      $this->filterPageAndOrder($items_query);
+      $nids = $items_query->execute();
+      $nodes = Node::loadMultiple($nids);
+      if (!empty($nodes)) {
+        foreach ($nodes as $node) {
+          $response_data['items'][] = $this->getItem($node);
+        }
+      }
+    }
+    $response = new JCMSRestResponse($response_data, Response::HTTP_OK, ['Content-Type' => $this->getContentType()]);
+    $response->addCacheableDependencies($nodes);
+    return $response;
+  }
+
+  /**
+   * Takes a node and builds an item from it.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $node
+   *
+   * @return array
+   */
+  public function getItem(EntityInterface $node) {
+    /* @var Node $node */
+    $item = $this->processDefault($node);
+
+    // Impact statement is optional.
+    if ($node->get('field_impact_statement')->count()) {
+      $item['impactStatement'] = $this->fieldValueFormatted($node->get('field_impact_statement'));
```

Comment:
looks duplicated with the item resource. We could share the code snippet with a small `JobAdvertPresenter` (Java-like name). Perhaps it is even more general and usable from multiple content types

Reply:
@giorgiosironi there are some redundancies here but when I get around to addressing the slow queries I can do that clean up then. The implementation here is consistent with that of other rest resources.